src/sage/schemes/elliptic_curves/ell_generic.py

```diff
@@ -175,15 +173,46 @@ def __init__(self, K, ainvs, category=None):
 
         self.__divpolys = ({}, {}, {})
 
-        # See #1975: we deliberately set the class to
-        # EllipticCurvePoint_finite_field for finite rings, so that we
-        # can do some arithmetic on points over Z/NZ, for teaching
-        # purposes.
-        if isinstance(K, sage.rings.abc.IntegerModRing):
-            self._point = ell_point.EllipticCurvePoint_finite_field
-
     _point = ell_point.EllipticCurvePoint
 
+    def assume_base_ring_is_field(self, flag=True):
+        r"""
+        Set a flag to pretend that this elliptic curve is defined over a
+        field while doing arithmetic, which is useful in some algorithms.
+
+        The flag affects all points created while the flag is set. Note
+        that elliptic curves are unique parents, hence setting this flag
+        may break seemingly unrelated parts of Sage.
```

Comment:
```suggestion

        .. WARNING::
        
            The flag affects all points created while the flag is set. Note
            that elliptic curves are unique parents, hence setting this flag
            may break seemingly unrelated parts of Sage.
```

Reply:
Breaking other parts of Sage seems bad, so I suggest adding a WARNING block for emphasis